### DISCUSSION TITLE
How to extend the functionality of docsVersionDropdown

### DISCUSSION THREAD

**Discussion by andreipope** [facebook/docusaurus#8057] [Q&A]
Hi,

I'm using Docusaurus to build a documentation website that will include an API Reference section rendered using [Stoplight Elements](https://github.com/stoplightio/elements). The problem I've stumbled upon is that Stoplight Elements doesn't support SSR and every time you try to switch the version it will jump to the home page. I'd like to know if it's possible to wrap and extend the functionality of the version dropdown so that users could select the version of the reference section that they want to see.

Thank you!

**Comment by slorber** [comment]
I would need to see a POC to understand the problem.

If Stoplight gets reloaded on each page navigation (pathname updated, version updated, i18n locale switched...), then you should have a way to tell it to restart on a given page, using querystring, hash, localstorage or whatever. If it's not possible, I don't know how it could be integrated into Docusaurus or anything else.

Never heard ot Stoplight before, what did you choose it instead of one of the existing community openapi plugins?
- https://github.com/rohit-gohri/redocusaurus
- https://github.com/PaloAltoNetworks/docusaurus-openapi-docs
- https://github.com/cloud-annotations/docusaurus-openapi

Note it's also possible to build something custom with MDX and React, see:
- https://www.courier.com/docs/reference/send/message/
- https://github.com/trycourier/docs/blob/main/docs/reference/send/message.mdx
- https://github.com/trycourier/docs/tree/main/src/components/ApiReference
- https://www.courier.com/blog/how-we-built-our-documentation/

**Reply by andreipope** [reply]
Hi,

Thanks for your prompt response. We mainly decided to use Stoplight because it seemed like it required fewer customizations in terms of UI and functionalities.
You can find a POC here: [https://twelvelabs-documentation-v2-git-demo-twelve-labs.vercel.app](https://twelvelabs-documentation-v2-git-demo-twelve-labs.vercel.app).
We'd like to know if it's feasible to integrate the version dropdown with Stoplight Elements or if we would be better off looking at other plugins or even building our custom elements using MDX as you suggested.

I appreciate your taking the time to look into this. 

Best,
Andrei.

**Reply by slorber** [reply]
This is a bug I noticed recently in the version dropdown: search/hash is not preserved when switching version. I'll fix this soon and backport in 2.2

As long as deep linking works, it should be possible to get spotlight work as it's based on hash-routing:
- https://twelvelabs-documentation-v2-git-demo-twelve-labs.vercel.app/api-reference#/operations/listIndexes
- https://twelvelabs-documentation-v2-git-demo-twelve-labs.vercel.app/1.0/api-reference#/operations/listIndexes

**Reply by andreipope** [reply]
Awesome! Thank you so much for looking into it.

**Reply by slorber** [reply]
https://github.com/facebook/docusaurus/pull/8059
